drivers/i2s/i2s_stm32_sai.c

```diff
@@ -427,10 +433,15 @@ static int i2s_stm32_sai_configure(const struct device *dev, enum i2s_dir dir,
 		hsai->Init.MckOutput = SAI_MCK_OUTPUT_DISABLE;
 	}
 
-	if (cfg->mclk_div_enable) {
+	if (cfg->mclk_div == (enum mclk_divider)MCLK_NO_DIV) {
 		hsai->Init.NoDivider = SAI_MASTERDIVIDER_ENABLE;
 	} else {
 		hsai->Init.NoDivider = SAI_MASTERDIVIDER_DISABLED;
```

Comment:
Tbh, this is confusing but may be totally right. Can you confirm this is correct ?